transport_nantes/mobilito/views.py

```diff
@@ -436,3 +443,41 @@ def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         context = self.get_context_data(**kwargs)
         self.check_view_permission(context["mobilito_session"])
         return self.render_to_response(context)
+
+
+@csrf_protect
+def report_session(request: HttpRequest, **kwargs) -> HttpResponse:
+    """Report a session as inappropriate"""
+
+    if request.method == 'POST':
+        if not request.user.is_authenticated:
+            return HttpResponse(status=401)
+        session_sha1 = kwargs['session_sha1']
+        try:
+            session_object = get_object_or_404(Session, session_sha1=session_sha1)
+        except Http404:
```

Comment:
This deserves an explanation why you believe this can't possibly happen in the normal course of affairs.  (I believe you're right, but it needs to be documented.)

Reply:
Just in case someone uses curl to post, not very likely I agree, but at least the event is handled

Reply:
CSRF protect

Reply:
And I agree with you that it's highly unlikely and that it's reasonable to 4xx if it happens.  But you _do_ need to explain clearly your reasoning in the code.

Reply:
Hmm right, the CSRF protect will prevent the post from originating from somewhere else.
Should I get rid of this try/except block altogether ? It might confuse readers with a very unlikely event

Reply:
No, you should **document** what you have done.